.github/workflows/spell-check.yaml

```diff
@@ -0,0 +1,16 @@
+name: Spell Check
```

Comment:
Looks like we already have a misspell tool in this repo. What's the difference between this and https://github.com/open-telemetry/opentelemetry-collector/blob/main/Makefile#L125-L132 ?